pkg/engine/virtualmachinescalesets.go

```diff
@@ -69,9 +69,7 @@ func CreateMasterVMSS(cs *api.ContainerService) VirtualMachineScaleSetARM {
 	}
 
 	if hasAvailabilityZones {
-		virtualMachine.Zones = &[]string{
-			"[parameters('availabilityZones')]",
-		}
+		virtualMachine.Zones = &masterProfile.AvailabilityZones
```

Comment:
Why this change?

Reply:
Because this is bad:

```
Azure Error: InvalidTemplate
Message: Deployment template parse failed: 'Error reading string. Unexpected token: StartArray. Path '[0]'.'.
```

Reply:
The real answer is the params evaluates to an array.  So it ends up being an array inside of an array (a pointer to), which isn't parseable.

Reply:
This change is good to go.